src/app/containers/AssignmentTemplateEditForm.tsx

```diff
@@ -0,0 +1,54 @@
+import * as React from 'react';
+import {
+    reduxForm,
+    ConfigProps
+} from 'redux-form';
+import { default as AssignmentForm, AssignmentFormProps } from '../components/AssignmentForm';
+import {
+    SheriffAssignment,
+    RecurrenceInfo
+} from '../api/index';
+import { editAssignmentTemplate } from '../modules/assignments/actions';
+import { default as FormSubmitButton, SubmitButtonProps } from '../components/FormElements/SubmitButton'
+import { connect } from 'react-redux';
+import { RootState } from '../store';
+
+
+// wrapping generic assignment form in redux-form
+const formConfig: ConfigProps<any, AssignmentFormProps> = {
+    form: 'EditAssignmentTemplate',
+    onSubmit: (values: {id:number, assignment: SheriffAssignment, recurrenceInfo: RecurrenceInfo[] }, dispatch, props) => {
+        let updatedAssignmentTemplate = Object.assign({}, { ...values });
+        dispatch(editAssignmentTemplate(updatedAssignmentTemplate));
+    }
+};
+
+export interface AssignmentTemplateEditFormProps extends AssignmentFormProps{
+    id: number;
+}
+
+const mapStateToProps = (state: RootState, props: AssignmentTemplateEditFormProps) => {
+    if (state && state.assignments && state.assignments.templates) {
+        const initialTemplate = state.assignments.templates[props.id];
+        return {
+            initialValues: initialTemplate, 
+            workSectionId: initialTemplate.assignment.workSectionId
```

Comment:
Do you still need to pass `workSectionId` since it's part of the **initialValues** ?